documentation/correctness-previous.adoc

```diff
@@ -0,0 +1,130 @@
+They could very well reside in another serialization.
+That pertains specifically to references to language entities of the LionCore builtins language, which is assumed to be imported/available implicitly.
+If all links do resolve, we say the serialization chunk is _self-contained_.
+
+=== Meta-referential
+
+All the metamodels specified in the serialization chunk should be _made available_ to the client('s deserializer) in the form of a runtime representation that the deserializer can reconstruct from the serialization.
+**Note** that this is a condition that pertains to the deserializing client, rather than on the serialization chunk.
+Incorrectness could be partially recovered from by partially deserializing into "`reflective`" nodes.
+
+=== Meta-structural
+
+The seralization matches the metamodels - in particular:
+
+* Classifier identifiers (tuples with a `key` value) are uniquely resolvable to a concept in a metamodel.
+Incorrectness is to be reported in terms of the ID of the serialized and the key of the concept which can not be resolved.
+* For every object corresponding to a serialized node:
+** For every entry of the `properties`, `children`, or `references` arrays, the `key` in the `property`, `containment`, resp. `reference` sub object is the key of a feature of the concept pointed to by the `concept` identifier.
+Incorrectness is to be reported in terms of the ID of the serialized node, the feature type (property/containment/reference), the index of the offending entry, and the key which can not be resolved.
+** For every entry of a `properties` array, the value of the `value` property conforms to the `type` specified by the `Property` feature corresponding to the key.
+
+Incorrectness is to be reported in terms of the ID of the serialized node, the feature type (property/containment/reference), the index of the offending entry, and the key which can not be resolved.
+"`To resolve a key`" means that we can find a concept or feature in some metamodel with that key, and with (exactly-)matching version.
+Incorrectness is to be reported as a model with references to nodes, as well as an appropriate error message.
+Incorrectness could be partially recovered from by partially deserializing into "`reflective`" nodes, leaving the unresolvable meta-references as-is.
+
+
+== Model (post-deserialization/in-memory)
+
+=== Structure-referential
+
+All links (on M1/instance-level) resolve.
+Note that this doesn't require any knowledge of the metamodels, as it just involves nodes' IDs.
+This means that the serialization chunk doesn't need to be correct in the meta-referential nor in the meta-structural sense!
+It should be alway possible to deserialize the serialization chunk into "`reflective`" nodes, which could be useful in its own right.
+
+**TODO:** describe when and how links should be resolved, etc.
+
+=== Constraints
+
+No constraint violations.
```

Comment:
What kind of constraints do you have in mind? Are these language-specific constraints (e.g., not having two methods with the same signature, or have at least one constructor, or do not have public methods in a private class)?

Reply:
Gave an example in the main document.